micrometer-core/src/test/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetricsTest.java

```diff
@@ -51,9 +51,9 @@ void cacheExposesMetricsForHitMissAndEviction() throws Exception {
         cache.put("user3", "Third User");
         cache.put("user4", "Fourth User");
 
-        assertThat(registry.find("c.requests").tags("result", "hit").tags(userTags).value(Count, 1.0).meter()).isPresent();
-        assertThat(registry.find("c.requests").tags("result", "miss").tags(userTags).value(Count, 2.0).meter()).isPresent();
-        assertThat(registry.find("c.evictions").tags(userTags).value(Count, 2.0));
+        assertThat(registry.mustFind("c.requests").tags("result", "hit").tags(userTags).functionCounter().count()).isEqualTo(1.0);
```

Comment:
Here is the new recommended way to assert metric values.

When calling the type (`functionCounter()` in this line) it checks for the name, tags, and type. If it isn't present the error message is: `Unable to locate a meter named 'c.requests' with Tags:[result:hit,userTagKey:userTagValue] of type io.micrometer.core.instrument.FunctionCounter`

So summary of changes:
* `find` and `mustFind` are the ways to look for metrics. 
* The metric type methods (`timer()`, `gauge()`, etc) now return the actual type instead of an `Optional` and throw an exception if the metric is not found (in the case of `mustFind`) or `null` (in the case of `find`)
* All tests are updated to leverage this simpler API.